### ISSUE TITLE
Fine-tuning with different normalization statistics

### ISSUE THREAD

**Issue by brianhenn** [ai2cm/ace#226]
Currently fine-tuning via `ExistingStepperConfig` reads the `Normalizer` from state on disk, meaning that fine-tuning is assumed to be done with a dataset with the same statistics as the original training. It would be nice to loosen this restriction, since to me fine-tuning doesn't imply identical dataset statistics (and presumably the AMIP and climSST statistics aren't quite the same). We could add an optional `NormalizationConfig` for the `ExistingStepperConfig`, for example, and read in new stats from the fine-tuning dataset.

**Comment by oliverwm1** [comment]
Maybe should discuss offline—but the model weights are tied to a particular set of normalization statistics, so I don't think it makes sense to change the normalization statistics for fine-tuning. Some thought may be required to get a set of normalization statistics that work well for both the AMIP and climSST datasets, but my hunch is that it doesn't matter too much since these datasets are not so different. For climate change experiments this could be a much more significant issue.

**Comment by brianhenn** [comment]
Agree that this probably is not actually a big deal for AMIP vs climSST. 

[Here](https://github.com/ai2cm/explore/blob/master/brianh/2023-11-08-FME-AMIP-normalization/2023-11-08-FME-AMIP-normalization.ipynb) is a notebook showing the differences in the centering and scaling of the variables. There are some differences in the scaling magnitude for the upper-level water variables (I think this is because Spencer tried to reduce the trends in specific_total_water_0 by starting the AMIP simulations from the end of the climSST simulations, though it looks like it the variability of specific_total_water_1 got bigger). There is also slightly more variability in precipitation, total water path + tendency, etc, as might be expected from SST variability, but it's not huge.

![scaling_diff](https://github.com/ai2cm/ace/assets/40010792/92de663a-254d-4282-9192-b032f7613653)

**Comment by brianhenn** [comment]
In the short term I agree it makes sense to use the climSST statistics in the control AMIP-from-scratch training for the clean comparison with the current code. But just leaving this here for a discussion, I think at some point we should try both (and hybrid) normalization statistics. I couldn't find much from quick googling; ChatGPT 3.5 says that the fine-tuning dataset's statistics should be used, but ChatGPT 4.0 says to try both and see what happens:grinning:.

**Comment by gideonite** [comment]
I think that from the ML perspective, the normalization statistics are part of the parameters of the model. If we are doing fine-tuning anyway and show the model a different dataset, I think it's fine in turn to change the normalization procedure. The question is what to do at inference time. Do we have to look up the normalization values for each particular dataset? Do we ever expect to run on our model on datasets that we have not been able to fine-tune over?

In general, given a held out test set, I think it's totally reasonable to normalize the data, run through the model, and denormalize the data using the values that were computed on the fly. @brianhenn

**Comment by brianhenn** [comment]
Closing this as stale given subsequent progress by others in fine-tuning
